roles/elasticsearch/defaults/main.yml

```diff
@@ -5,17 +5,12 @@ es_data_dir: "{{ rock_data_dir }}/elasticsearch"
 es_cluster_name: rocknsm
 es_node_name: "{{ inventory_hostname }}"
 es_network_host: >-
-  {%- if (groups['elasticsearch']|union(groups['logstash'])|union(groups['kibana']))| count > 1 -%}
+  {%- if (groups['elasticsearch']|union(groups['logstash'])|union(groups['kibana']))| length > 1 -%}
   _site:ipv4_
   {%- else -%}
   _local:ipv4_
   {%- endif -%}
-es_interface: >-
-    {%- if (groups['elasticsearch']|union(groups['logstash'])|union(groups['kibana']))| count > 1 -%}
-    {{ ansible_hostname }}
-    {%- else -%}
-    localhost
-    {%- endif -%}
+es_interface: "{{ ansible_hostname }}"
```

Comment:
I know the logic is painful, but it has to match the logic in `es_network_host` because during a rolling cluster update, we have to use the `url` module to access the individual host because we restart one host a time. And to be generalized for when we're running Elasticsearch on loopback, `ansible_hostname` won't be reachable on `:9200`.